code/implementation/simulator/agent-connection/src/main/java/io/cattle/platform/agent/connection/simulator/AgentConnectionSimulator.java

```diff
@@ -32,6 +32,10 @@ public AgentConnectionSimulator(ObjectManager objectManager, Agent agent, List<A
 
     @Override
     public Event execute(Event event) {
```

Comment:
This causes the simulated agent to behave as if it were disconnected. I trigger it in SimulatorStartStopProcessor below